### PR TITLE
parse config: should ignore null values

### LINKED ISSUE #182: TypeError: Cannot convert undefined or null to object
Since #177 my application throws the error:
```
TypeError: Cannot convert undefined or null to object
/app/api/node_modules/loopback-boot/lib/executor.js in null.<anonymous> at line 403
```

It is because I parse myself env variables for datasource config. The results of the parsing may contains `null` values but:
```js
typeof null === 'object'
```
And so:
```js
Object.keys(null)
```
throws: **TypeError: Cannot convert undefined or null to object**

The `null` values were things like `port` etc...
I updated my application for giving default port and so avoid null value but it should certainly be fixed.

### PR DESCRIPTION
**Pull request by LoicMahieu**
Fix #182

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by davidcheung** [comment]
@slnode ok to test

File: lib/executor.js [hunk 1]
```diff
@@ -400,7 +400,7 @@ function getUpdatedConfigObject(app, config, opts) {
         interpolated[configKey] = value.map(interpolateVariables);
       } else if (typeof value === 'string') {
         interpolated[configKey] = getConfigVariable(value);
-      } else if (typeof value === 'object' && Object.keys(value).length) {
+      } else if (typeof value === 'object' && value !== null && Object.keys(value).length) {
```
Comment (davidcheung) [inline lib/executor.js]: Please wrap line so it doesn't exceed 80 length.

File: test/executor.test.js [hunk 2]
```diff
@@ -543,6 +543,23 @@ describe('executor', function() {
...
+      boot.execute(app, simpleMiddlewareConfig('routes',
+        { nested: { info: { path: '${restApiRoot}', some: null }}}
+      ));
+
+      supertest(app).get('/').end(function(err, res) {
+        if (err) return done(err);
+        expect(res.body.nested).to.eql({
+          info: {
+            path: app.get('restApiRoot'),
+            some: null
```
Comment (davidcheung) [inline test/executor.test.js]: please add comma-dangle so eslint passes

**Comment by davidcheung** [comment]
@LoicMahieu Thanks for the patch! 
@bajtos PTAL

**Comment by LoicMahieu** [comment]
Rebased!

**Comment by bajtos** [comment]
@LoicMahieu landed, thank you for the contribution! I have released your patch in `loopback-boot@2.18.1`.

**Pull request merged**